### ISSUE TITLE
Lazy string evaluation does not work in log message

### ISSUE THREAD

**Issue by vojtechsokol** [gluetool/gluetool#101]
Consider the following module stub:

```python
class SomeModule(gluetool.Module):
    name = 'somemodule'
    ...

    def execute(self):
        self.info('Lazy string evaluation test: %d', 42)
```

One would expect that this would produce the following entry in the log:
`[08:40:08] [+] [somemodule] Lazy string evaluation test: 42`

However the actual result is:
`[08:40:08] [+] [somemodule] Lazy string evaluation test: %d`

**Comment by happz** [comment]
There is a small, teeny-tiny problem with Python 2 syntax :/ Our logging methods have some additional keyword parameters, e.g. `sentry` with all the type annotations involved, their signatures look like this:

```python
def info(self, msg, exc_info=None, extra=None, sentry=False):   # type: ignore
    # type: (str, Optional[ExceptionInfoType], Optional[Dict[str, Any]], bool) -> None
```

The additional positional arguments are not accepted, on purpose, for several reasons:

* we never used this approach, we somehow started to use `.format()` for all our formatting work, including logging, it's just consistent across all uses.
* Python 2 vs Python 3 difference when it comes to `*args` position: `def foo(a, *args, b=None, **kwargs)` is valid in Python 3, but not in Python 2. `def foo(a, b=None, *args, *kwargs)` is valid in both, but `foo('abc %s', 'baz')` would assign `baz` to `b`, not to `args`. Which is not good given we want those extra and explicit keyword args and type checking and so on.
* lazy evaluation has its advantages, but it never did for us, I'd say. In our workflows, we *always* emit logging messages, even those below the logging level, because we use `gluetool` pretty much exclusively with the debug file enabled, i.e. our messages are always evaluated.

The actual blocker is our (large) module base not supporting Python 3. We have plans to get rid of Python 2 support and move on, but the priority was never high enough, compared to other goals :/

My suggestion would be to use `.format()` as we do, although I am aware of the exception the old-style formatting got when used with logging, and it indeed is commonly used. I'm not really sure I want to invest a non-trivial amount of work into making this work right now, as we're "stuck" with Python 2 - when we make our modules compatible with Python 3, and begin actually using Python 3 to power our pipelines, then absolutely, yes, this could happen, easily.

@thrix care to join in?

**Comment by vojtechsokol** [comment]
Well then there is problem with libraries that might be used by the gluetool, if their code uses the old-style formatting (which is common). E.g when `jaeger` is used for tracing, its logging is worthles, because [this line](https://github.com/jaegertracing/jaeger-client-python/blob/28f672b239b7598f2a1096eb384c4a4c20299ba3/jaeger_client/reporter.py#L73) outputs this to the log:
```
[17:09:52] [D] [tracing] Reporting span %s
```

But this is just a FYI, I'm in no position to tell if it's a serious issue. So feel free to close this issue, I have no problem to use the new formatting for the logging strings.

**Comment by thrix** [comment]
Yep, I agree with @happz that we should fix this, but it is not a priority now. Thanks for reporting @vojtechsokol, I was personally not aware of this issue. Let's keep this issue open and address once we move to Python3 only, which is coming ....
